sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_distributed_tracing.py

```diff
@@ -0,0 +1,582 @@
+                        query_text = obj.get("query")
+                    elif "queryText" in obj:
+                        query_text = obj.get("queryText")
+                    # parameters may be list of {name, value}
+                    p = obj.get("parameters") or obj.get("params") or obj.get("parametersList")
+                    if isinstance(p, list):
+                        for entry in p:
+                            if isinstance(entry, dict) and "name" in entry:
+                                key = entry.get("name")
+                                if isinstance(key, str) and key.startswith("@"):
+                                    params[key] = entry.get("value")
+                                else:
+                                    params[str(key)] = entry.get("value")
+                    # fallback parse a single 'parameters' dict
+                    if isinstance(p, dict):
+                        for k, v in p.items():
+                            params[str(k)] = v
+                # if body is a string SQL, treat as query
+                elif isinstance(obj, str):
+                    query_text = obj
+            except Exception:
+                # not JSON; treat entire body as query text if looks like SQL
+                trimmed = s.strip()
+                if trimmed.upper().startswith("SELECT") or trimmed.upper().startswith(
+                        "FROM") or "WHERE" in trimmed.upper():
+                    query_text = trimmed
+    except Exception as ex:
+        logger.debug("Error extracting query and parameters: %s", ex, exc_info=True)
+    return query_text, params
+
+
+class CosmosDistributedTracingPolicy(DistributedTracingPolicy):
+    """
+    Cosmos-specific distributed tracing policy.
+
+    - Overrides on_request/on_response/on_exception and does not call parent implementations.
+    - Uses only header names defined on `HttpHeaders`.
+    - Records `azure.cosmosdb.response.sub_status_code` when available.
+    - Degrades gracefully when OpenTelemetry or optional exceptions are unavailable.
+    """
+
+    def __init__(self, tracing_attributes: dict = {}, **kwargs: Any) -> None:
+        user_attrs = tracing_attributes if tracing_attributes else kwargs.pop("tracing_attributes", {})
+        required_attrs = {_CosmosTracingConstants.DB_SYSTEM_NAME: _CosmosTracingConstants.DB_SYSTEM_VALUE,
+                          _CosmosTracingConstants.DB_OPERATION_NAME: None}
+        merged_attrs = {**required_attrs, **user_attrs}
+        kwargs["tracing_attributes"] = merged_attrs
+        super().__init__(**kwargs)
+
+    def _safe_set_opt_in_attribute(self, span: "AbstractSpan", key: str, value: Optional[Any]) -> None:
+        """
+        Set opt-in attributes on the span if they are part of the OPT_IN_ATTRIBUTES list.
+        """
+        span.add_attribute(key, value)
+
+    def on_request(self, request: PipelineRequest[HTTPRequestType]) -> None:
+        ctxt = request.context.options
```

Comment:
The first thing we need to do on request is [confirm that tracing is enabled](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/azure/core/pipeline/policies/_distributed_tracing.py#L119-L121) - otherwise performance will suffer considerably in non-tracing scenarios.